packages/components/src/DataGrid/DataGridRow/DataGridRow.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  type ChangeEvent,
+  type MouseEvent,
+  type ReactNode,
+  useState,
+} from 'react';
+
+import { Checkbox } from '../../Checkbox';
+import { TableCell, TableRow } from '../../Table';
+import { Tooltip } from '../../Tooltip';
+import type { DataGridRowOptions } from '../types';
+
+export type DataGridRowProps<Data extends Record<string, unknown>> = {
+  keyId: keyof Data;
+  row: Data;
+  activeRowId?: string;
+  selectable?: boolean;
+  selectedRows?: Array<Data>;
+  options?: DataGridRowOptions;
+  children: ReactNode;
+  onSelectRow: (row: Data) => (event: ChangeEvent<HTMLInputElement>) => void;
+  onRowClick?: (row: Data) => void;
+};
+
+export const DataGridRow = <Data extends Record<string, unknown>>({
+  keyId,
+  activeRowId,
+  selectable,
+  selectedRows,
+  options,
+  row,
+  children,
+  onSelectRow,
+  onRowClick,
+}: DataGridRowProps<Data>) => {
+  const [isVisibleTooltip, setVisibleTooltip] = useState<boolean>(false);
+
+  const { isDisabled, disabledReason } = options || {};
+  const rowId = row[keyId] as string;
+  const checked =
+    selectable &&
+    Boolean(selectedRows?.find((selectedRow) => selectedRow[keyId] === rowId));
+
+  const handleMouseMove = (event: MouseEvent<HTMLElement>) => {
+    const element = event.target as HTMLElement;
+
+    if (isDisabled && element.tagName !== 'TD') {
+      setVisibleTooltip(false);
+    }
+  };
+
+  const handleRowClick = (row: Data, isDisabled?: boolean) => () => {
+    if (isDisabled) {
+      return undefined;
+    }
+
+    onRowClick?.(row);
+  };
+
+  return (
+    <Tooltip
+      title={isDisabled && disabledReason}
+      followCursor
+      arrow={false}
+      open={isVisibleTooltip}
+      onOpen={() => setVisibleTooltip(true)}
```

Comment:
https://kaluga-astral.github.io/style-guide/docs/rules/react/logic#%D0%B7%D0%B0%D0%BF%D1%80%D0%B5%D1%89%D0%B5%D0%BD%D0%B0-%D1%80%D0%B5%D0%B0%D0%BB%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D1%8F-%D0%BE%D0%B1%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D1%87%D0%B8%D0%BA%D0%BE%D0%B2-%D0%B2-jsx